core/frontend/src/components/common/PasswordInput.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <v-text-field
+    @input="emitInput"
+    @keyup.enter="emitSubmit"
+    :value="value"
+    :append-icon="show_password ? 'mdi-eye' : 'mdi-eye-off'"
+    :type="show_password ? 'text' : 'password'"
+    @click:append="show_password = !show_password"
+    label="Password"
+    hide-details="auto"
+    class="pa-2"
+  />
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+export default Vue.extend({
+  name: 'PasswordInput',
+  props: {
+    value: {
+      type: String,
+      default: '',
+    },
+  },
+  data () {
+    return {
+      show_password: '',
```

Comment:
why not a a boolean?

Reply:
No idea why this is a string. It was a boolean for sure. More strange is the fact that this works. Fixing.

Reply:
How typescript failed to identify the wrong use of string here ??

Reply:
What happened what that, when using `show_password ?`, the empty string was being cast to a boolean, returning `false`, and on the first call to `show_password = !show_password`, it was turned into a *de facto* boolean. All the logic works fine and the operations are valid.

The problem I see is that the Typescript linter was unable to see that a variable was having its type altered on runtime. This happened because this toggle operation happened inside the `<template>`. When I move the operation to a method, inside the `<script>` tag, the linter is able to catch the type-change mistake and forces me to declare it initially as a boolean.

This enforces that we should avoid any operation inside the template, and use `computed` and `methods` whenever possible.